### ISSUE TITLE
Testing/QA Plan

### ISSUE THREAD

**Issue by kcole16** [nearprotocol/near-wallet#460]

**Comment by kcole16** [comment]
@vgrichina @Patrick1904 @marcinbodnar Thoughts?

**Comment by vgrichina** [comment]
@kcole16 I totally agree we should start adding end 2 end tests for critical user journeys. Cypress looks like best options these days (@Patrick1904 already added some stuff https://github.com/nearprotocol/near-wallet/pull/491). 

> Merges to master are deployed to staging, not production

I agree with this in sense of how to run stuff on MainNet. MainNet production release process IMO also needs to include review of generate JS bundle diff compared to previous release (to decrease possibility of unnoticed npm module backdoors). We have to do a bit over the top stuff because of security concerns.

However for testnet I think it's important to keep living dangerously, as it creates far bigger responsibility for merging production-ready changes and urgency to fix immediate issues. Third-party developers also always do far more realistic testing than we can simulate in staging. |

We however need to start monitoring testnet web apps using BugSnag, Sentry, etc – this will allow to catch many problems earlier.

Also generally I am not sure why deploying merged changes to staging would provide much bigger certainty for us than testing individual PR in isolation like we do today. We already have staging environment spun up for every unmerged PR.

> Manual QA cycle covering user journeys required before promoting to production

Overview of what I think is critical to test:
https://www.notion.so/nearprotocol/Betanet-Release-Draft-bf7048bd397d4fe78347ebf3e82fe874#733020436cb84bdbb72e4d4c6ebe9e92
